### PR TITLE
bind in /etc/origin/node for non-master monitoring to be able to talk…

### PR DESCRIPTION
**Pull request by joelddiaz**
… with master

### CODE REVIEW

File: roles/oso_host_monitoring/templates/oso-rhel7-host-monitoring.service.j2 [hunk 1]
```diff
@@ -62,6 +62,8 @@ ExecStart=/usr/bin/docker run --name {{ osohm_host_monitoring }}
            -v /etc/origin/master/master.etcd-client.crt:/etc/origin/master/master.etcd-client.crt \
            -v /etc/origin/master/master.etcd-client.key:/etc/origin/master/master.etcd-client.key \
            -v /etc/origin/master/master-config.yaml:/etc/origin/master/master-config.yaml \
+{% elif hostvars[inventory_hostname]['ec2_tag_host-type'] == 'node' %}
```
Comment (twiest) [inline roles/oso_host_monitoring/templates/oso-rhel7-host-monitoring.service.j2]: We should start using the oo_ vars as they'll be cross cloud.

So, `oo_hosttype`.

**Comment by twiest** [comment]
:+1:

**Comment by detiber** [comment]
I have a few questions related to this (mostly around improvements that can be made in the future):
- Why does the monitoring on the nodes need to talk to the master?
- What permissions are needed for this communication? It seems like using the master/node kubeconfigs would probably be granting way too much access for monitoring.
- Why not create and use a service account for master/kubelet communication for monitoring purposes?

**Comment by joelddiaz** [comment]
@detiber , there was a desire for the non-master nodes to be able to test and report on the health of the docker registries. The minimum permissions needed are simply to get the IP address of the docker-registry service and POD endpoint IPs.

I'll discus with @twiest about using the service account, and what the implications might be for these dedicated clusters (will customers see and/or object to these accounts?).

**Comment by joelddiaz** [comment]
@detiber I spoke with @twiest . Basically, these monitoring containers are root/super-privileged and are supposed to run the equivalent of root on the bare metal of the instance/node. Using a service account is something we can definitely look into. And it might make sense to consider this change when we move the monitoring containers to be run/deployed by OpenShift itself.

No concerns about customers as they wouldn't have visibility into the existence of the service account/monitoring containers from the customer's namespace.

**Comment by detiber** [comment]
@joelddiaz thanks for the additional info. Using service accounts will definitely be much easier once the containers are running within OpenShift, since then the generated secrets will be provided within the container automatically.

Using serviceaccounts where possible for accessing the api is generally good practice, especially if the admin privileges aren't needed.

**Pull request merged**